server/src/main/java/io/crate/rest/action/RestBulkRowCountReceiver.java

```diff
@@ -43,7 +43,10 @@ class RestBulkRowCountReceiver extends BaseResultReceiver {
     @Override
     public void setNextRow(Row row) {
         rowCount = (long) row.get(0);
-        failure = (Throwable) row.get(1);
+        // Can be an optimized bulk request with only 1 bulk arg/operation which carries only 1 column (row count).
+        if (results.length > 1) {
+            failure = (Throwable) row.get(1);
+        }
```

Comment:
Instead of using the `result.length` as the detection, we could also use a `try/catch` block to catch possible IndexOutOfBoundsException`. The later maybe safer (we currently *expect*  this behaviour if the results length is 1), but afaik more expensive. WDYT?

Reply:
length check seems fine to me